### ISSUE TITLE
Compare training loss of train.track baseline vs `embed_inputs`

### ISSUE THREAD

**Issue by purzelrakete** [feldberlin/wavenet#4]
## What

Reproduce the train.track baseline, then change `embed_inputs=True` and 

## Hypothesis

I would like to know that `embed_inputs` works properly, and that performance on a real world audio snippet is improved quite a bit vs the baseline. 

Anecdotally, modelling the inputs as categorical variables yields significantly better results. It's not clear to me why this must be true. The inputs of the digital sound file are ordinal in nature, since PCM data is quantised. However the underlying signal is continuous. Consequently it seems more natural to model this using continuous inputs. This is also often done in the image domain, where images are normalised before being passed into networks. I don't recall seeing any vision work where images are one hot encoded before being passed into the network. 

On the other hand, the two main oss wavenet implementations both seem to have started with continuous inputs and then moved to categorical encodings, since this massively improved the performance. It even seems that the google wavenet implementation did something similar, as the original authors stated privately. This seems consistent with their previous papers, e.g. PixelCNN, where everything was treated as a categorical variable.

## Results

Ran an  [experiment](https://wandb.ai/purzelrakete/feldberlin-wavenet/runs/21ei0tqc?workspace=user-purzelrakete) using a Track dataset. I had to use mono encoding, since there's still a bug with stereo embedded inputs. Regardless, it seems clear that the model is able to converge massively faster. The training loss is now down to almost zero, which is exactly what I would want to see when trying to overfit a single track.  

Caveat: I was not able to generate a track from this run yet, due to a fast generation bug. 

## Acceptance Criteria

- [x] reproduced train.track baseline
- [x] ran the treatment: embedded inputs
- [x] compared and wrote up results

**Comment by purzelrakete** [comment]
Treatment  is still at 4.6 loss at epoch 13, where the baseline was at 2.5 loss at epoch 13. Looks like there's a serious problem here. 

## Theories:

This is happening due to a problem in folding stereo. Input embeddings worked really well for the mono sinusoids, and also for the tiny dataset. The main difference I can see with the track dataset is that it's stereo. 

## Questions

- Are there any other substantial differences between train.track and train.sines?

## TODO

- [ ] write and pass a test comparing fast and simple generation when using `embed_inputs`
- [ ] run another experiment, but squash to mono

**Comment by purzelrakete** [comment]
Mono [experiment](https://wandb.ai/purzelrakete/feldberlin-wavenet/runs/21ei0tqc?workspace=user-purzelrakete)
